### PR TITLE
Add RubyGems repository proxy documentation

### PR DESCRIPTION
**Pull request by cd-work**

### CODE REVIEW

File: docs/artifact_repositories/rubygems.md
Comment (maxrake) [inline docs/artifact_repositories/rubygems.md:1]: This new page should be added to `docs/artifact_repositories/about.md`.

File: docs/artifact_repositories/rubygems.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+A mirror for Ruby's default repository can be configured using `bundle config`.
+We define a fallback timeout of `9999` to prevent Ruby from silently falling
+back to the official registry should Phylum's mirror go down.
+
+```sh
+bundle config set --global mirror.https://rubygems.org https://rubygems.phylum.io
+bundle config set --global mirror.https://rubygems.org.fallback_timeout 9999
+```
+
+After setting up the mirror, you also need to provide phylum's group and API key
```
Comment (maxrake) [inline docs/artifact_repositories/rubygems.md:38]:
```suggestion
After setting up the mirror, you also need to provide Phylum's group and API key
```

**Review by maxrake — CHANGES_REQUESTED** [review]

File: docs/artifact_repositories/rubygems.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+> ⚠️ **WARNING** ⚠️
+>
+> Do not accidentally save your token into your shell history.
+
+A blocked package will show up in `bundle` output as missing:
+
+```text
+Could not find gem 'bad_gem' with platform 'x86_64-linux' in rubygems repository https://rubygems.org/ or installed locally.
```
[the remark this answers is not in the document]
Comment (cd-work) [reply docs/artifact_repositories/rubygems.md:52]: Yeah I was wondering if I should point out that this is what is supposed to happen, but I figured people will figure that out themselves when they see it locally and compare it.

[the remark this answers is not in the document]
Comment (cd-work) [reply docs/artifact_repositories/rubygems.md:52]: It surprised me too tbh.

**Review by maxrake — APPROVED** [review]
LGTM

File: docs/artifact_repositories/rubygems.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxrake) [reply docs/artifact_repositories/rubygems.md:52]: Yeah...that should work...and we can make it more clear later if there are questions.

**Pull request merged**